src/test/java/co/rsk/federate/signing/hsm/client/HSMClientVersion2BTCTest.java

```diff
@@ -418,37 +417,25 @@ public void sendBlockHeadersChunks_keepPreviousChunkLastItem_false() throws Json
 
     @Test
     public void updateAncestorBlock_ok() throws HSMClientException, JsonRpcException {
-        when(jsonRpcClientMock.send(any(JsonNode.class))).thenReturn(buildResponse(0));
-        when(jsonRpcClientMock.send(hsmClientProtocol.buildCommand("blockchainState", 2)))
-                .thenReturn(buildResponse(0, "state", buildStateResponse(false)));
-
-        client.setMaxChunkSizeToHsm(2);
-
-        BlockHeader blockHeader = mock(BlockHeader.class);
-        when(blockHeader.getEncoded(true, false)).thenReturn(new byte[]{});
-        List<BlockHeader> blockHeaders = Arrays.asList(blockHeader, blockHeader, blockHeader);
-
-        client.updateAncestorBlock(new UpdateAncestorBlockMessage(blockHeaders));
-        ArgumentCaptor<JsonNode> captor = ArgumentCaptor.forClass(JsonNode.class);
-        verify(jsonRpcClientMock, times(3)).send(captor.capture());
-        List<JsonNode> capturedArguments = captor.getAllValues();
-        Assert.assertEquals("blockchainState", capturedArguments.get(0).get("command").asText());
-        Assert.assertEquals("updateAncestorBlock", capturedArguments.get(1).get("command").asText());
-        Assert.assertEquals("updateAncestorBlock", capturedArguments.get(2).get("command").asText());
+        // hsm version 2
+        test_for_hsm_version(2);
+        // hsm version 3
+        test_for_hsm_version(3);
+        // hsm version 4
+        test_for_hsm_version(4);
     }
 
-    @Test
-    public void updateAncestorBlock_hsm_version_3() throws HSMClientException, JsonRpcException {
+    public void test_for_hsm_version(int hsmVersion) throws HSMClientException, JsonRpcException {
```

Comment:
```suggestion
    public void test_updateAncestorBlock(int hsmVersion) throws HSMClientException, JsonRpcException {
```
I think that's more explanatory

Reply:
Resolved